spec/unit/provider/oneview_volume_template_synergy_spec.rb

```diff
@@ -0,0 +1,105 @@
+################################################################################
+# (C) Copyright 2016-2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+require 'spec_helper'
+require_relative '../../support/fake_response'
+require_relative '../../shared_context'
+
+provider_class = Puppet::Type.type(:oneview_volume_template).provider(:synergy)
+resourcetype = OneviewSDK::VolumeTemplate
```

Comment:
```ruby
api_version = login[:api_version] || 200
resource_name = 'VolumeTemplate'
resourcetype = Object.const_get("OneviewSDK::API#{api_version}::Synergy::#{resource_name}") unless api_version < 300
```
And it could likely be put bellow the describe provider_class